cargo/flatpak-cargo-generator.py

```diff
@@ -74,6 +74,28 @@ async def get_remote_sha256(url):
     return sha256.hexdigest()
 
 
+def find_toml(git_repo_dir):
+    if os.path.isfile(os.path.join(git_repo_dir, 'Cargo.toml')):
+        return load_toml(os.path.join(git_repo_dir, 'Cargo.toml')), '.'
+    # If Cargo.toml isn't found in repository root, scan subdirectories
+    toml_list = []
+    for entry in os.scandir(git_repo_dir):
+        if entry.is_dir():
+            toml_path = os.path.join(git_repo_dir, entry.name, 'Cargo.toml')
+            if os.path.isfile(toml_path):
+                toml_list.append((toml_path, os.path.relpath(entry.path, git_repo_dir)))
+    if len(toml_list) == 1:
+        return load_toml(toml_list[0][0]), toml_list[0][1]
+    if len(toml_list) > 1:
+        raise NotImplementedError(
+            f'Multiple Cargo.toml files found in {git_repo_dir}\n'
+            'Please report this error and the link to the affected repisitory here:\n'
+            'https://github.com/flatpak/flatpak-builder-tools/issues'
+            )
+    if not toml_list:
```

Comment:
This `if` clause is redundant, since we have either `raise`d or `return`ed before it's reached.

Reply:
`len(toml_list)` can still be `0`, that's what that line is for.
I can write it as `len(toml_list) == 0` to make that clear.

Reply:
What I'm saying is that `len(toml_list)` can be *only* equal to zero at this point, since we've already checked if it's exactly 1 or more than 1. So, there is no need to check for equality to zero explicitly.